packages/view-util/lib/filters.js

```diff
@@ -0,0 +1,43 @@
+import { view, viewBuilder } from '@view-builder/core/ns.js'
+import { rdf, schema } from '@tpluscode/rdf-ns-builders'
+
+export function generateLookupSources(filter) {
+  filter
+    .deleteOut(view.operation)
+    .addOut(view.operation, view.Eq)
+
+  filter
+    .deleteOut(view.argument)
+    .addOut(view.argument, filter.out(viewBuilder.filterTermSet))
+
+  filter.addOut(view.dimension, (dimension) => {
+    dimension.addOut(view.as, schema.inDefinedTermSet)
+    dimension.addOut(view.from, (from) => {
+      from.addOut(view.source, (source) => {
+        source.addOut(rdf.type, view.LookupSource)
```

Comment:
What is the role of lookupSource?

Reply:
To join observations with other data. Originally, I think it was for non-RDF but here the library uses it to create a join with the term set (instead of `view:In`)

@bergos will explain better